common/utils.cpp

```diff
@@ -659,5 +659,181 @@ void setFruPresence(const std::string& fruObjPath, bool present)
+    uint32_t size{};
+    const pldm_pdr_record* record{};
+    std::vector<std::vector<uint8_t>> pdrs;
+
+    do
+    {
+        record = pldm_pdr_find_record_by_type(repo, PLDM_STATE_EFFECTER_PDR,
+                                              record, &outData, &size);
+
+        if (record)
+        {
+            auto pdr = reinterpret_cast<pldm_state_effecter_pdr*>(outData);
+            if (pdr)
+            {
+                auto compositeEffecterCount = pdr->composite_effecter_count;
+                auto possible_states_start = pdr->possible_states;
+
+                for (auto effecters = 0x00; effecters < compositeEffecterCount;
+                     effecters++)
+                {
+                    auto possibleStates =
+                        reinterpret_cast<state_effecter_possible_states*>(
+                            possible_states_start);
+                    auto setId = possibleStates->state_set_id;
+                    auto possibleStateSize =
+                        possibleStates->possible_states_size;
+
+                    if (pdr->entity_type == entityType)
+                    {
+                        std::vector<uint8_t> effecter_pdr(&outData[0],
+                                                          &outData[size]);
+                        pdrs.emplace_back(std::move(effecter_pdr));
+                        break;
+                    }
+                    possible_states_start += possibleStateSize + sizeof(setId) +
+                                             sizeof(possibleStateSize);
+                }
+            }
+        }
+    } while (record);
+
+    return pdrs;
+}
+
+std::vector<pldm::pdr::EffecterID>
+    findEffecterIds(const pldm_pdr* pdrRepo, uint8_t tid, uint16_t entityType,
+                    uint16_t entityInstance, uint16_t containerId)
+{
+    std::vector<uint16_t> effecterIDs;
+
+    auto pdrs = getStateEffecterPDRsByType(tid, entityType, pdrRepo);
+    for (const auto& pdr : pdrs)
+    {
+        auto effecterPdr =
+            reinterpret_cast<const pldm_state_effecter_pdr*>(pdr.data());
+        if (effecterPdr)
+        {
+            auto compositeEffecterCount = effecterPdr->composite_effecter_count;
+            auto possible_states_start = effecterPdr->possible_states;
+
+            for (auto effecters = 0x00; effecters < compositeEffecterCount;
+                 effecters++)
+            {
+                auto possibleStates =
+                    reinterpret_cast<const state_effecter_possible_states*>(
+                        possible_states_start);
+                auto possibleStateSize = possibleStates->possible_states_size;
+                if (entityType == effecterPdr->entity_type &&
+                    entityInstance == effecterPdr->entity_instance &&
+                    containerId == effecterPdr->container_id)
+                {
+                    uint16_t id = effecterPdr->effecter_id;
+                    effecterIDs.emplace_back(std::move(id));
```

Comment:
why this std::move operation is required here?

Reply:
std::move in used in this function and many others in the same file when emplace_back is used to place a variable in a vector. I think the main purpose in general is to tell the compiler place this file in this vector or place it at the end and reuse the memory for id variable because we are in a for loop. std::move is used in most functions in this file. FYR - https://stackoverflow.com/questions/3413470/what-is-stdmove-and-when-should-it-be-used